src/aptsources_cleanup/util/pkg.py

```diff
@@ -20,6 +20,9 @@ def check_integrity(pkg, paragraphs, debug_fail=0, *,
 
 	try:
 		with FileDescriptor(md5sums_file) as md5sums_fd:
+			if isinstance(md5sums_fd, int):
+				raise EnvironmentError(
+					'Got error code {} when opening {}'.format(md5sums_fd, md5sums_file))
```

Comment:
Thanks for bringing that bug to my attention. Fixed in f397f3aa3b246b6121b4cb9bfdec3c5402f8341d which obviates this (commented) patch.